scripts/common/smoketest.go

```diff
@@ -0,0 +1,155 @@
+package common
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"net"
+	"os"
+	"os/exec"
+	"os/signal"
+	"path"
+	"path/filepath"
+	"strconv"
+	"syscall"
+)
+
+type SmokeTest []*exec.Cmd
+
+type SmokeTestOpts struct {
+	LogDir     string
+	RepoDir    string
+	NumWorkers int
+	Compile    func() error
```

Comment:
Oh cool, I see what you're doing. It's a shame that golang makes it hard to mix functions and data, because an interface just to hold data feels weird, but a struct to hold functions also feels weird...